### DISCUSSION TITLE
JDK 17 support the target byte code is still Java 11

### DISCUSSION THREAD

**Discussion by tchikh411** [oracle/opengrok#4513] [General]
In the release notes for version 1.12.17, OpenGrok added the JDK 17 support but it also mentions the target byte code is still Java 11. Can you elaborate on this? Is it safe to run opengrok in JDK17 in production?

**Comment by vladak** [comment]
The Java 11 target byte code was chosen that deployments which cannot upgrade Java for some reason can still run OpenGrok. Running with Java 17 should be fine, the standard/implicit notices apply.

**Comment by tchikh411** [comment]
Thanks @vladak
